### ISSUE TITLE
False positive `Unreachable pattern` on const in range

### ISSUE THREAD

**Issue by ortem** [intellij-rust/intellij-rust#3577]
```Rust
const MAX: usize = 10;

fn main() {
    let v = vec![1, 2, 3];
    match v.len() {
        1 => { println ! ("1"); }
        2...MAX => { println ! ("2...MAX"); } // "Unreachable pattern"
        _ => { println ! ("other"); }
    }
}
```

**Comment by angusholder** [comment]
I'm getting `Unreachable pattern` lint warnings from IntelliJ on the `EOR` branch in the following code

    fn calculate(opcode: u32, op1: u32, op2: u32) -> u32 {
        const AND: u32 = 0b0000;
        const EOR: u32 = 0b0001;
    
        match opcode {
            AND => op1 & op2,
            EOR => op1 ^ op2,
            _ => unimplemented!()
        }
    }

Does #3658 fix this or should I file a separate issue? @Undin

**Comment by Undin** [comment]
@angusholder
> Does #3658 fix this

Yeap, so you don't need to file new issue
